lib/Dialect/FIRRTL/FIRRTLOps.cpp

```diff
@@ -2083,6 +2122,10 @@ LogicalResult ConnectOp::verify() {
   if (failed(checkConnectFlow(*this)))
     return failure();
 
+  // Check constraints on RefType.
```

Comment:
Yes-- it'd be nice if was only `strictconnect`, since the requirements are indeed strict, but `connect` is needed for supporting references to unsized base types.  Types still have to match, though.